### PR TITLE
chore: consolidate CI/CD tooling, contributor docs, and beta-release setup

### PR DESCRIPTION
**Pull request by michaeln31**
## Overview

This brings the CI/CD and tooling work from the `michaeln31` fork back onto upstream. Everything
here was previously reviewed and merged incrementally on the fork's `develop`, and arrives as a
single fast-forward because `main` is a direct ancestor — so there are **no code conflicts and no
functional changes to the integration**. It's tooling, docs, and metadata only.

The goal is to give the repo a modern, self-checking pipeline: every push and PR is linted,
formatted, tested across Python versions, and validated the way Home Assistant and HACS expect — so
regressions are caught automatically rather than by hand.

## What's included

**Continuous integration**
- **Linting/formatting:** flake8 replaced with **ruff**, wired into both CI and a
  `.pre-commit-config.yaml` so issues are caught locally before they're pushed.
- **Testing:** CI runs a **Python 3.13/3.14 matrix** with coverage and uv caching, plus **hassfest**
  and **HACS** validation (the checks HA/HACS use to accept an integration).
- **Type checking:** a report-only **mypy** baseline to burn typing debt down over time.
- **Hardening:** least-privilege workflow permissions, concurrency cancellation, and SHA-pinned
  actions.
- **Maintenance:** split `requirements-dev.txt`, **dependabot** (pip + actions), a **devcontainer**
  for local parity with CI, and a seeded `quality_scale.yaml` to track HA Quality Scale progress.
- A single stable **`CI Success`** status check that stays constant even as the job matrix changes,
  so branch protection has one reliable target to require.

**Docs & metadata**
- **`README`:** fixed the broken CI badge (`python-app.yml` → `ci.yml`), corrected the HACS
  custom-repo URL, and reconciled the version heading/changelog to `0.7.1`.
- **`CONTRIBUTING.md`** (new): documents the contribution flow, Conventional Commits, and releases.
- **Ownership:** added `@michaeln31` to `codeowners` + a `.github/CODEOWNERS` file.

## Releases & the optional `develop` branch

`main` stays the primary, stable trunk — contributions normally target it directly. A `develop`
branch is **optional**: it's only used when a change should ship as a `-beta.N` prerelease for
testing before landing on `main`. A follow-up PR will add **release-please** (seeded at `0.8.0`) to
automate both the stable cuts on `main` and the beta cuts on `develop`. CI already triggers on both
branches so either path is covered.

## Notes for review
- The `graphql_client.py` diff is **pure ruff reformatting** — no logic change.

### CODE REVIEW

**Comment by michaeln31** [comment]
@toby200 here's the initial suggested changes. It adds some "best practice" ci/cd patterns. This PR is only foundational stuff, there are a few more improvements needed, but this is a good start.

The CI / Typecheck is expected to fail for a little longer, the check is a later requirement but is allowed to fail until i can do a refactor. I've kept this PR functional, no changes to the logic.

If you have any questions, please let me know.

I would prefer to merge this myself, when I can monitor. I would also choose to squash these commits

File: custom_components/andersen_ev/manifest.json [hunk 1]
```diff
@@ -1,13 +1,13 @@
...
+  "iot_class": "cloud_polling",
   "issue_tracker": "https://github.com/HA-AndersenEV/hassio-andersen-ev/issues",
-  "dependencies": [],
-  "codeowners": ["@lwsrbrts", "@toby200"],
   "requirements": ["pycognito", "gql[aiohttp]", "boto3"],
-  "config_flow": true,
-  "iot_class": "cloud_polling",
-  "version": "0.7.1",
-  "icon": "mdi:ev-station"
+  "version": "0.7.1"
```
[the remark this answers is not in the document]
Comment (toby200) [reply custom_components/andersen_ev/manifest.json:12]: Ignore this one!

File: requirements-dev.txt [hunk 2]
```diff
@@ -0,0 +1,8 @@
+-r requirements.txt
+homeassistant==2026.1.3
```
[the remark this answers is not in the document]
Comment (michaeln31) [reply requirements-dev.txt:2]: This is correct and tracked in future commit. The homeassistant==2026.1.3 pin isn't new in this PR, it already existed in requirements.txt and this change only relocates it into requirements-dev.txt. Deciding whether to test against the declared 2025.3.0 floor (or bump that floor to match what we test) is a support-policy question rather than CI/CD plumbing, so I'd rather not fold it into a "no logic changes" tooling PR. I'll open a follow-up to either add a minimum-version leg to the test matrix or reconcile the HACS minimum.

File: custom_components/andersen_ev/services.yaml [hunk 3]
```diff
@@ -21,9 +21,6 @@ get_device_info:
       example: "1b6f9e38a4d72c0f5e831649"
       selector:
         text: {}
-  response:
-    name: Device Information
-    description: Returns detailed information about the selected Andersen EV charger
```
[the remark this answers is not in the document]
Comment (michaeln31) [reply custom_components/andersen_ev/services.yaml:24]: I believe this one should stay as-is. The response: blocks were removed intentionally when adding hassfest and HACS validation, because response is not a valid key in Home Assistant's services.yaml schema and hassfest rejects it. Service response support is declared in code via supports_response=True (which we do set for both services in __init__.py), not documented in services.yaml. Re-adding the response: section would reintroduce a hassfest failure, which is exactly the validation this PR is wiring up.

File: custom_components/andersen_ev/services.yaml [hunk 4]
```diff
@@ -36,9 +33,6 @@ get_device_status:
       example: "1b6f9e38a4d72c0f5e831649"
       selector:
         text: {}
-  response:
-    name: Device Status
-    description: Returns detailed status for the selected Andersen EV charger
```
[the remark this answers is not in the document]
Comment (michaeln31) [reply custom_components/andersen_ev/services.yaml:36]: Same as previous.

File: .github/dependabot.yml [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+          - "*"
+
+  - package-ecosystem: "pip"
+    directory: "/"
+    target-branch: "develop"
+    schedule:
+      interval: "weekly"
+    commit-message:
+      prefix: "chore"
+      include: "scope"
```
[the remark this answers is not in the document]
Comment (michaeln31) [reply .github/dependabot.yml:20]: This was originally intentional, so Dependabot would flow through the beta/prerelease branch but i think it would cause less headaches to go with this suggestion.

**Review by toby200 — APPROVED** [review]
